### DISCUSSION TITLE
req.files can be null in [@types/express-fileupload]

### DISCUSSION THREAD

**Discussion by motsgar** [DefinitelyTyped/DefinitelyTyped#61604] [Issues with a @types package] [answered]
The type of req.files is defined wrong as it can actually be null.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b6235d0da4053464b414513c0511fe02e25cc0eb/types/express-fileupload/index.d.ts#L14

https://github.com/richardgirges/express-fileupload/blob/4f81fc822d385e3e404a3266a76c103f543c3a2a/lib/processMultipart.js#L28

My program crashed to req.files being null after I check if it is undefined

[a comment by an automated account is not shown]

**Reply by peterblazejewicz** [reply]
I'd avoid strict type-checking like in terse C# kind syntax, instead of using relaxed JS expression evaluaton, so:
```ts
req.files && req.files.length;
// or
req.files?.length
```
so TS can do type checking behind, e.g.:
```js
(_a = req.files) === null || _a === void 0 ? void 0 : _a.length;
```
(non-native nullable), but if you want to improve types, please submit PR

**Reply by BendingBender** [reply]
Made a PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61835

**Comment by motsgar** [comment, marked as answer]
The PR seems to be merged so considering as resolved
